Docs/private funcs 371

ritesh006 opened the pull request:
Doxyzen style docs added for private function

Commit:
docs: add VS Code IDE setup doc (#473)

Commit:
docs: add Doxygen comments for private element helpers

Commit:
docs(private): addressed the issues

goatshriek commented on include/private/element.h:
```diff
@@ -25,7 +25,23 @@
+ * Iterates the parameters of an element and executes the following block only
+ * for parameters whose name matches the provided name.
+ *
+ * **Thread Safety: MT-Safe req:element-stable**
```
I can't find anything regarding the `req` safety attribute remark in the attributes man page. Where did you get this remark from?

goatshriek commented on include/private/element.h:
```diff
@@ -37,9 +53,49 @@ for( i = 0; i < ( ELEMENT )->param_count; i++ ) {   \
+ * must not be called without the element being locked by the caller.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * This function is not intended for use in signal handlers.
```
Provide a reason that this is AS-Unsafe, particularly since it is marked as AC-Safe.

goatshriek reviewed (changes requested):
Thanks for putting this together! Just a few changes/questions for you to address.

ritesh006 commented:
Addressed feedback:

Replace req: with MT-Safe race:element; note no concurrent changes to element->params/param_count.

Macro: AS-Unsafe (lock+strcmp), AC-Unsafe (lock).

locked_get_param_by_index: AS-Safe / AC-Safe + caller must hold element lock.

Docs only; no behavior change.

ritesh006 commented:
Hi @goatshriek I Have addressed your comment please review this

goatshriek commented on include/private/element.h:
```diff
@@ -25,7 +25,23 @@
+ * element->param_count are not modified during iteration. Each param is locked
+ * only for the duration of the strcmp() comparison.
+ *
+ * **Async Signal Safety: AS-Unsafe lock strcmp**
```
`strcmp` is not a recognized unsafe attribute (see [this list](https://www.gnu.org/software/libc/manual/html_node/Unsafe-Features.html)). Remove it from the header and leave it in the description below.

goatshriek reviewed (changes requested):
Just one more chance and this should be ready to go.

goatshriek commented on include/private/element.h:
```diff
@@ -25,7 +25,23 @@
+ *
+ * **Async Cancel Safety: AC-Unsafe lock**
+ * May block at mutex operations, which are cancellation points.
+*/
```
Correct this indentation error by adding a space before the `*/`, which is likely what is causing the failure in the single file build (due to the way it parses the headers).

goatshriek reviewed (changes requested):
One more minor change before this can be accepted, due to some failures the CI runs noted.

ritesh006 commented:
here is something due to branch is not merged.

goatshriek commented:
Thanks for your contribution and patience in working through the requested changes!
